artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientProducerImpl.java

```diff
@@ -219,6 +233,8 @@ private void doSend(SimpleString sendingAddress,
          // In case we received message from another protocol, we first need to convert it to core as the ClientProducer only understands core
          ICoreMessage msg = msgToSend.toCore();
 
+         msg.putStringProperty(PRODUCER_ID, new SimpleString(uuid.toString()));
```

Comment:
The getters always return a String via a toString call, and this always creates a new SimpleString (and calls toString on the UUID) on every send...might be better to store the SimpleString rather than UUID and e.g use the toString as the getter returns and the key to remove from the sessionContext, avoids repeated recreation?

Reply:
fixed